spyder/plugins/projects/tests/test_plugin.py

```diff
@@ -120,5 +127,33 @@ def test_open_project_uses_visible_config(projects, tmpdir, value):
     assert projects.dockwidget.isVisible() == value
 
 
+def test_set_get_project_filenames(projects, tmpdir, mocker):
+    """
+    Test that the currently opened files in the Editor are saved and loaded
+    correctly to and from the project config when the project is closed and
+    then reopened.
+
+    Regression test for Issue #8375
+    """
+    opened_files = ['file1', 'file2', 'file3']
+    mocker.patch.object(
+        projects.main.editor, 'get_open_filenames', return_value=opened_files)
+
+    # We mock os.path.isfile so that we do not have to
+    # actually create the files on the disk.
+    mocker.patch('spyder.plugins.projects.api.os.path.isfile',
```

Comment:
Is this what we're using now instead of ``monkeypatch``?

Reply:
I don't know, that was just my understanding from the previous files but its up to @ccordoba12

Reply:
I don't have a problem with using any approach.